### PR TITLE
2.0-preview instance update bug: checking keys

### PR DESCRIPTION
**Pull request by mkazlauskas**
Fixed a bug where having
```html
<app-localstorage-document key="key1"></app-localstorage-document>
<app-localstorage-document key="key2"></app-localstorage-document>
```

and updating data in one of them would update data of the other one.

### CODE REVIEW

File: app-localstorage/app-localstorage-document.html [hunk 1]
```diff
@@ -221,8 +222,8 @@
         }
       },
 
-      __setStorageValue: function(key, value) {
-        this.storage.setItem(this.key, JSON.stringify(this.data));
+      __setStorageValue: function(key, value = null) {
```
Comment (e111077) [inline app-localstorage/app-localstorage-document.html]: Not supported in Safari 9. perhaps:

`value = value === undefined ? null : value;`

File: test/app-storage-compatibility-suite.html [hunk 2]
```diff
@@ -221,27 +221,35 @@
...
+            storageOther.transactionsComplete,
           ]);
         });
 
         teardown(function() {
           context.unsetValue(storageOne);
-          return context.awaitUpdate(storageTwo).then(function() {
-            expect(storageTwo.data).to.be.eql(storageOne.data);
-          });
+          context.unsetValue(storageOther);
```
Comment (e111077) [inline test/app-storage-compatibility-suite.html]: This change and the `key` feature is specific to `app-localstorage-document` and some other implementations / extensions of `app-storage-behavior` may not have a feature like this. I suggest moving this test to `test/app-localstorage/app-localstorage-document.html` alongside the compatibility suite.

Comment (mkazlauskas) [inline test/app-storage-compatibility-suite.html]: Perhaps someone who set the tests up could do that?

Comment (e111077) [inline test/app-storage-compatibility-suite.html]: Unfortunately, you're modifying these tests in this PR, and I don't think this is an implementation that I can merge into this branch in this state. Additionally, you are the one that knows your code best! 👨‍💻

**Comment by mkazlauskas** [comment]
Replaced with #115

**Pull request closed without merging**